plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloat.hs

```diff
@@ -318,11 +334,12 @@ floatBackLets term fTable =
             TyAbs a n k <$> (floatLam (n^.PLC.theUnique) =<< go tBody)
         -- Unfloatable-let anchor, increase depth
         Let a r bs@(representativeBindingUnique -> letU) tIn -> local (+1) $
-            -- note that we do not touch the recursivity of an unfloatable-let
-            Let a r
-                <$> (floatRhs letU =<< traverseOf (traversed.bindingSubterms) go bs)
-                -- act the same as lam/Lam: float right inside
-                <*> (floatLam letU =<< go tIn)
+            -- note that we do not touch the recursivity of the unfloatable-let
+            bindingGrpToLet r
```

Comment:
I think it's clearer to do it like this (if you make the other change I suggest to `bindingGrpToLet`): you've already used the recursivity of the let term itself when you create the binding groups below.
```suggestion
            bindingGrpToLet NonRec
```